pkg/riff/commands/application_status_test.go

```diff
@@ -0,0 +1,215 @@
+/*
+ * Copyright 2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package commands_test
+
+import (
+	"testing"
+
+	duckv1alpha1 "github.com/knative/pkg/apis/duck/v1alpha1"
+	buildv1alpha1 "github.com/projectriff/system/pkg/apis/build/v1alpha1"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+
+	"github.com/projectriff/cli/pkg/riff/commands"
+	rifftesting "github.com/projectriff/cli/pkg/testing"
+)
+
+func TestApplicationStatusOptions(t *testing.T) {
+	table := rifftesting.OptionsTable{
+		{
+			Name: "invalid resource",
+			Options: &commands.ApplicationStatusOptions{
+				ResourceOptions: rifftesting.InvalidResourceOptions,
+			},
+			ExpectFieldError: rifftesting.InvalidResourceOptionsFieldError,
+		},
+		{
+			Name: "valid resource",
+			Options: &commands.ApplicationStatusOptions{
+				ResourceOptions: rifftesting.ValidResourceOptions,
+			},
+			ShouldValidate: true,
+		},
+	}
+
+	table.Run(t)
+}
+
+func TestApplicationStatusCommand(t *testing.T) {
+	defaultNamespace := "default"
+	appName := "test-app"
+	imageTag := "registry.example.com/repo:tag"
+	gitRepo := "https://example.com/repo.git"
+	gitMaster := "master"
+
+	table := rifftesting.CommandTable{
```

Comment:
missing a test case for when there are no conditions. This is common if the reconciler is not running.